src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,196 @@
+            String index = part[1].trim();
+
+            try {
+                return Integer.parseInt(index) - 1;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Invalid format. Please use integers only.");
+            }
+        } else {
+            throw new DukeException("Invalid format. Please use 'delete <index>'.");
+        }
+    }
+
+    /**
+     * Parses the mark command and returns the index of the task to be marked as done.
+     *
+     * @return The index of the task to be marked as done.
+     * @throws DukeException If the command is in an invalid format.
+     */
+    public int parseMark() throws DukeException {
+        String[] part = command.split(" ");
+
+        if (part.length == 2) {
+            String index = part[1].trim();
+            try {
+                return Integer.parseInt(index) - 1;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Invalid format. Please use integers only.");
+            }
+        } else {
+            throw new DukeException("Invalid format. Please use 'mark <index>'.");
+        }
+    }
+
+    /**
+     * Parses the unmark command and returns the index of the task to be marked as not done.
+     *
+     * @return The index of the task to be marked as not done.
+     * @throws DukeException If the command is in an invalid format.
+     */
+    public int parseUnmark() throws DukeException {
+        String[] part = command.split(" ");
+
+        if (part.length == 2) {
+            String index = part[1].trim();
+            try {
+                return Integer.parseInt(index) - 1;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Invalid format. Please use integers only.");
+            }
+        } else {
+            throw new DukeException("Invalid format. Please use 'unmark <index>'.");
+        }
+    }
+
+    /**
+     * Parses the event command and returns the corresponding Event.
+     *
+     * @return The Event created based on the event command.
+     * @throws DukeException If the command is in an invalid format.
+     */
+    public Event parseEvent() throws DukeException {
+        String input = command.substring(Duke.Command.EVENT.name().length()).trim();
+        int byIndex = input.indexOf(" /from ");
+        if (byIndex != 0 && byIndex != -1) {
+            String description = input.substring(0, byIndex).trim();
+            String fromTo = input.substring(byIndex + 6).trim();
+            String[] part = fromTo.split(" /to ", 2);
+            if (part.length == 2) {
+                String from = part[0].trim();
+                String to = part[1].trim();
+                return new Event(description, from, to);
+            }
+        }
+        throw new DukeException("Invalid format. Please use 'event <description> /from <datetime> /to <datetime>'.\n" +
+                " Datetime format: <yyyy-MM-dd HH:mm>.");
```

Comment:
Perhaps the `+` operator could be moved to the front of the second line to better follow the coding standard? I noticed this in other places too.